File: app/src/main/java/no/joharei/flixr/api/models/Photosets.java

package no.joharei.flixr.api.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class Photosets {

    private int cancreate;
    private int page;
    private int pages;
    private int perpage;
    private int total;
    @SerializedName("photoset")
    private List<Photoset> photosets = new ArrayList<>();

    public List<Photoset> getPhotosets() {
        return photosets;
    }
}

File: app/src/main/java/no/joharei/flixr/api/models/Photosets.kt
package no.joharei.flixr.api.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class PhotosetsResponse(val photosets: Photosets)

@JsonClass(generateAdapter = true)
data class Photosets(
    val page: Int,
    val pages: Int,
    val perpage: Int,
    val total: Int,
    @Json(name = "photoset")
    val photosets: List<Photoset>
)